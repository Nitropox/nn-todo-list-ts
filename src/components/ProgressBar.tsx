import React from "react";

export const ProgressBar: React.FC = () => {
  return <div></div>;
};
