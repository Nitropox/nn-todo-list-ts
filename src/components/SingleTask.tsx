import React from "react";

export const SingleTask: React.FC = () => {
  return <div></div>;
};
