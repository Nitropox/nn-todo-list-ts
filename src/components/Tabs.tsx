/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";

interface TabsProps {
  activeTab: number;
  switchTab: Function;
}

export const Tabs: React.FC<TabsProps> = ({ activeTab, switchTab }) => {
  return (
    <ul className="nav nav-tabs">
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 0 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(0)}
        >
          Wszystkie
        </a>
      </li>
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 1 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(1)}
        >
          Wykonane
        </a>
      </li>
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 2 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(2)}
        >
          Niewykonane
        </a>
      </li>
    </ul>
  );
};
