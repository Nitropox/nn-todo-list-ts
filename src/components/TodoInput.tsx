import React, { useState, useContext } from "react";
import { Context as TodosContext } from "../context/tasksReducer";

export const TodoInput = () => {
  const [input, setInput] = useState("");
  const [isValid, setIsValid] = useState(true);
  const { addTask } = useContext(TodosContext);

  const submitHandler = (event: any): void => {
    event.preventDefault();
    if (!event.target.checkValidity()) {
      setIsValid(false);
    } else {
      setIsValid(true);
      addTask(+new Date(), input, () => setInput(""));
    }
  };

  return (
    <div className="row pt-2">
      <form
        className={`col needs-validation ${isValid ? "" : "was-validated"}`}
        noValidate
        onSubmit={submitHandler}
      >
        <div className="input-group mb-3">
          <input
            type="text"
            className="form-control"
            placeholder="Wprowadź zadanie"
            value={input}
            onChange={event => setInput(event.target.value)}
            maxLength={100}
            required
          />
          <div className="input-group-append">
            <button className="btn btn-primary rounded-right" type="submit">
              +
            </button>
          </div>
          <div className="valid-feedback">Pole wypełnione poprawnie</div>
          <div className="invalid-feedback">To pole nie moe być puste!</div>
        </div>
      </form>
    </div>
  );
};
