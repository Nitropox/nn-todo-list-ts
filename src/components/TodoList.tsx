import React from "react";

export const TodoList: React.FC = () => {
  return <div></div>;
};
