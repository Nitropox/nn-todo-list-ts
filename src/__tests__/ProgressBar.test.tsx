import { ProgressBar } from "../components/ProgressBar";
import React from "react";
import { shallow } from "enzyme";

describe("ProgressBar component", () => {
  it("Renders ProgressBar component", () => {
    expect(shallow(<ProgressBar progress="20%" />).length).toEqual(1);
  });

  it("Matches the snapshot", () => {
    expect(shallow(<ProgressBar progress="20%" />)).toMatchSnapshot();
  });
});
