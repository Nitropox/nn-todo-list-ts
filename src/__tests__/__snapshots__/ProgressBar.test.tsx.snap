// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component Matches the snapshot 1`] = `ShallowWrapper {}`;
